pulp_container/app/json_schemas.py

```diff
@@ -1,23 +1,32 @@
 from pulp_container.constants import (
-    ALLOWED_ARTIFACT_TYPES,
-    ALLOWED_BLOB_CONTENT_TYPES,
     BLOB_CONTENT_TYPE,
     MEDIA_TYPE,
     SIGNATURE_TYPE,
 )
 
 
 def get_descriptor_schema(
-    allowed_media_types, additional_properties=None, additional_required=None
+    allowed_media_types=None, additional_properties=None, additional_required=None
 ):
     """Return a concrete descriptor schema for manifests."""
 
+    media_type_config = {"type": "string"}
+    if allowed_media_types is not None:
+        media_type_config = {"type": "string", "enum": allowed_media_types}
+
     properties = {
-        "mediaType": {"type": "string", "enum": allowed_media_types},
+        "mediaType": media_type_config,
         "size": {"type": "number"},
         "digest": {"type": "string"},
         "annotations": {"type": "object", "additionalProperties": True},
-        "urls": {"type": "array", "items": {"type": "string"}},
+        "urls": {
+            "type": "array",
+            "items": {"type": "string"},
+            "format": "uri",
+            "pattern": "^https?://",
+        },
+        "data": {"type": "string", "contentEncoding": "base64"},
+        "artifactType": {"type": "string"},
```

Comment:
you forgot subject :grimacing:

Reply:
hum... not sure if I'm looking at the wrong schema, but I could not find the definition for `subject` in the descriptor:
https://github.com/opencontainers/image-spec/blob/main/descriptor.md#properties

Reply:
https://github.com/pulp/pulp_container/pull/1533/files#r1516647629